src/array.rs

```diff
@@ -21,6 +21,21 @@ impl<T: RuntimeType> Array<T> {
         Self::default()
     }
 
+    pub fn with_len(len: usize) -> Self {
+        unsafe {
```

Comment:
Can we move the unsafe block to only be around lines 27-33? The other lines are safe